modules/services/mako.nix

```diff
@@ -312,5 +311,17 @@ in {
       ${optionalBoolean "ignore-timeout" cfg.ignoreTimeout}
       ${optionalString "group-by" cfg.groupBy}
     '';
+
+    systemd.user.services.mako = {
+      Unit = {
+        Description = "Mako, lightweight notification daemon for Wayland";
+        After = [ "graphical-session-pre.target" ];
+        PartOf = [ "graphical-session.target" ];
+      };
+
+      Service = { ExecStart = "${pkgs.mako}/bin/mako"; };
+
+      Install = { WantedBy = [ "graphical-session.target" ]; };
```

Comment:
Please, add an option to make the "Install" target configurable like Kanshi:
https://github.com/nix-community/home-manager/blob/7e5fee4268f53be8758598b7634dff8b3ad8a22b/modules/services/kanshi.nix#L160-L166
https://github.com/nix-community/home-manager/blob/7e5fee4268f53be8758598b7634dff8b3ad8a22b/modules/services/kanshi.nix#L191

Reply:
Thanks for the fast review!

I am not quite happy with this change: do you suggest to use "sway-session.target" as the default or can this be "graphical-session.target"? As mako is not sway specific I think it should be the second one. This is also the case for the upstream systemd file [here](https://github.com/emersion/mako/blob/master/contrib/systemd/mako.service.in), which I just stumbled upon (Sidenote: is it possible to reuse the upstream service file or do we prefer to define them again in nix?). As mako depends on a graphical session and wayland compositors started as a service should bind to "graphical-session.target" I can't really think of a use case changing the targets here.

Also, is it semantically correct to use the same target for both "After" and "PartOf"? This doesn't seem right, but upstream and the kanshi service do it, dunst on the other hand has a different target for "After"...

The other changes seem good to me. As someone who doesn't use GitHub a lot: should I just force-push the changes to not clutter the commit history or should those changes be a new commit?

Reply:
1. You are right, the default should most likely be `graphical-session.target`.

2. We mostly prefer to use our own service definition since the executable path in the upstream service file often needs to be changed (if the service file is not generated during the build process).

3. The `Install.WantedBy` will automatically start the service when the specified target or service is started.
`PartOf` is only useful to stop services as stopping `graphical-session.target` will stop all services `PartOf` it.

4. I think using the `After` clause will only start the service once all other services (that are not `After`) in the chain of `graphical-session.target` are started, so effectively when the `graphical-session.target` has reached the state of "active" instead of "activating".

5. You don't have to force push while a review is ongoing, you are only required to force push when you squash all commits into one.
Most contributions I've seen lately would amend their commit and force-push.